libs/components/layout/src/lib/modules/box/box.component.ts

```diff
@@ -38,4 +47,17 @@ export class SkyBoxComponent {
    */
   @Input()
   public ariaRole: string | undefined;
+
+  @ViewChild('titleRef', { read: ElementRef })
+  public set boxHeaderRef(value: ElementRef | undefined) {
+    if (value?.nativeElement?.textContent) {
```

Comment:
This is onceable; the text content of the element can change after `boxHeaderRef` is set. What happens if you always set the descriptor regardless of the contents of the header element? Wouldn't the screen reader just skip the element's empty content or a missing element?

Reply:
What happens is it doesn't ready anything for this empty element, but it does read the other non-empty element which says "context menu for". The problem is we always render `.sky-box-header-content` in box even if they don't specify any of the content that gets ng-contented in there--so if they don't include a header, as it is currently implemented their context menu would read "context menu for"

Unfortunately I think the answer is to move the "check" down a level. Box has to set the content info, but only box-header knows if a header element was actually provided--which if one was, I think we can safely assume there is content in it. I say unfortunately because I think that might mean another layer of viewchild so the parent can get the info from the child, but I haven't actually had a chance to think that through yet

Reply:
We know if `sky-box-header` is present, though, and I think it's OK to assume that if that component is present, it has text content and that we should use its element ID for the content info. The consumer can always set the ARIA label directly on the context menu for the extreme edge case where `sky-box-header` is present but has no inner text.